### DISCUSSION TITLE
Is there an event when you click the new item button?

### DISCUSSION THREAD

**Discussion by mreinhart2805** [Megabit/Blazorise#3737] [Q&A]
Is there an event that is triggered when e.g. a new element is created.
So, I want to disable other form fields while creating or editing an element.
is that possible?

**Comment by David-Moreira** [comment]
Is this DataGrid related? Can you give us more context?

**Reply by mreinhart2805** [reply]
OMG. it's late Yes, it's about the DataGrid. I've searched but can't find anything about it. I would like to deactivate other form fields while the EditTemplates are displayed. I don't mean the form fields from the DataGrid itself. I have multiple forms on one page and i want disable some Buttons.

**Reply by David-Moreira** [reply]
So there's a few ways you could go about this I guess.

Having a simple flag that sets the disabled on every field that you wish seems like a good start.

Then you can:
- Set your own custom logic on a Button of yours and then use the DataGrid's `New` command to make the Datagrid go into New Item mode. You'd probably need to listen the `RowInserted` and `Cancel` Button to disable the flag again.
  - You can do these by using the `DataGridCommandColumn`... `ButtonRowTemplate` or just have your own buttons.

```
<NewCommandTemplate>
  <Button Color="Color.Success" Clicked="@( () => {disabled=true; domylogic(); context.Clicked(); })">@context.LocalizationString</Button>
</NewCommandTemplate>
<CancelCommandTemplate>
  <Button Color="Color.Secondary" Clicked="@(()=> {disabled=false; domycancellogic(); context.Clicked()})">@context.LocalizationString</Button>
</CancelCommandTemplate>
```

- Look at the Datagrid's `EditState` it tells you whether Datagrid is in new or edit mode

**Reply by mreinhart2805** [reply]
Ok, first of all thanks. The manual possibility to set the data grid in new / edit mode was not entirely clear to me. I think that will be enough. Thanks for the hint.
